src/lambda/models/samTemplates.ts

```diff
@@ -10,20 +10,21 @@ import { Runtime } from 'aws-sdk/clients/lambda'
 import { Set as ImmutableSet } from 'immutable'
 import { supportsEventBridgeTemplates } from '../../../src/eventSchemas/models/schemaCodeLangs'
 import { RuntimePackageType } from './samLambdaRuntime'
+import { getIdeProperties } from '../../shared/extensionUtilities'
 
-export const helloWorldTemplate = 'AWS SAM Hello World'
-export const eventBridgeHelloWorldTemplate = 'AWS SAM EventBridge Hello World'
-export const eventBridgeStarterAppTemplate = 'AWS SAM EventBridge App from Scratch'
-export const stepFunctionsSampleApp = 'AWS Step Functions Sample App'
+export const helloWorldTemplate = localize('AWS.samcli.initWizard.template.helloWorld.name', '{0} SAM Hello World', getIdeProperties().company)
+export const eventBridgeHelloWorldTemplate = localize('AWS.samcli.initWizard.template.helloWorld.name', '{0} SAM EventBridge Hello World', getIdeProperties().company)
+export const eventBridgeStarterAppTemplate = localize('AWS.samcli.initWizard.template.helloWorld.name', '{0} SAM EventBridge App from Scratch', getIdeProperties().company)
+export const stepFunctionsSampleApp = localize('AWS.samcli.initWizard.template.helloWorld.name', '{0} Step Functions Sample App', getIdeProperties().company)
 export const repromptUserForTemplate = 'REQUIRES_AWS_CREDENTIALS_REPROMPT_USER_FOR_TEMPLATE'
 
 export const CLI_VERSION_STEP_FUNCTIONS_TEMPLATE = '0.52.0'
 
 export type SamTemplate =
-    | 'AWS SAM Hello World'
-    | 'AWS SAM EventBridge Hello World'
-    | 'AWS SAM EventBridge App from Scratch'
-    | 'AWS Step Functions Sample App'
+    | typeof helloWorldTemplate
+    | typeof eventBridgeHelloWorldTemplate
+    | typeof eventBridgeStarterAppTemplate
+    | typeof stepFunctionsSampleApp
```

Comment:
Not sure the type system is the right way to model this. If sam templates are localized, this won't scale... and it is getting pretty messy already.

These template names are more like ids than localizable names. Instead of adding new localization names for this purpose would it make sense to change the logic to match on a regex, e.g. 

    export const helloWorldTemplate = /(AWS|Amazon) SAM Hello World/

Reply:
I'll go back and check what these were doing specifically but honestly I think it's best just to dump this type and use a string.